modules/bluefish/util/blue_velvet.cpp

```
@@ -20,67 +20,231 @@
 */
 
 #include "../StdAfx.h"
-
 #include "blue_velvet.h"
+#include <common/utf.h>	
+#include <core/video_format.h>
+#include <bluefish/interop/BlueVelvetCUtils.h>
 
-#include <common/utf.h>
+#if defined(__APPLE__)
+	#include <dlfcn.h>
+#endif
 
-#include <core/video_format.h>
+#if defined(_WIN32)
```

Throw exception using `CASPAR_THROW_EXCEPTION` instead of returning `false`.